### PR TITLE
457 replace request module

### PR DESCRIPTION
**Pull request by thomasschuiki**
This solves #457 .

I updated all functions using [request](https://github.com/request/request) to use [got](https://github.com/sindresorhus/got) instead. The functions basically had to be rewritten, but the logic remains the same.

All available tests are passing. :)

During testing I had some issues with the remote download locations not responding or taking a long time to repsond. This could probably improved by mocking the responses for development. 

I'm happy about any feedback!

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -414,30 +412,26 @@ function install(opts, cb) {
...
-    r.end();
+  function getDownloadStream(downloadUrl, cb) {    
+    (async () => {
+      const downloadStream = got.stream(downloadUrl)
+      downloadStream
+        .on("response",(res) => {
+          startedRequests += 1;                
+        })
+        .on("downloadProgress", ({ transferred, total, percent }) => {                                 
+          let chunk = (16*1024);
```
Comment (mgrybyk) [inline lib/install.js:423]: Let's move it to the top of the file

File: lib/install.js [hunk 2]
```diff
@@ -414,30 +412,26 @@ function install(opts, cb) {
...
-
-        cb(null, res);
-      })
-      .once("error", function (error) {
-        cb(new Error("Could not download " + downloadUrl + ": " + error));
-      });
-
-    // initiate request
-    r.end();
+  function getDownloadStream(downloadUrl, cb) {
```
Comment (mgrybyk) [inline lib/install.js:415]: Can you make the function async inatead?

File: package.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "start": "DEBUG=selenium-standalone:* ./bin/selenium-standalone install && DEBUG=selenium-standalone:* ./bin/selenium-standalone start",
     "test": "./bin/selenium-standalone install && mocha",
+    "test-install": "mocha -g 'should install'",
```
Comment (mgrybyk) [inline package.json:9]: Cab you remove this one?

File: test/default-downloads-test.js [hunk 4]
```diff
@@ -13,18 +13,17 @@ var opts = {
 };
 
 function doesDownloadExist(url, cb) {
-  var req = request.get(url);
-  req.on('response', function(res) {
-    req.abort();
-
-    if (res.statusCode >= 400) {
-      return cb('Error response code got from ' + url + ': ' + res.statusCode);
+  (async () => {
```
Comment (mgrybyk) [inline test/default-downloads-test.js:16]: Change the parent function to async

File: lib/install.js [hunk 5]
```diff
@@ -594,35 +582,27 @@ function isUpToDate(url, requestOpts, hash, cb) {
...
-            res.statusCode
-          )
-        );
+  (async () => {
+      try {
+          const res = await got(url, options);
+          if (res.statusCode === 304) {
+            return cb(null, true);
+          }
+          if(res.statusCode !== 200) {
```
Comment (mgrybyk) [inline lib/install.js:600]: I guess if status code is not 200 for throws an error so it cannot be ever reached

File: lib/check-started.js [hunk 6]
```diff
@@ -1,31 +1,30 @@
...
-
-    request(hub, function (err, res) {
-      if (err || res.statusCode !== 200) {
-        setTimeout(hasStarted, retryInterval);
-        return;
+  };
+  (async () => {
+      try {
+          const res = await got(seleniumStatusUrl, options);
+          if(res.statusCode !== 200) {
```
Comment (mgrybyk) [inline lib/check-started.js:22]: I guess if status code is not 200 for throws an error so it cannot be ever reached

File: lib/check-started.js [hunk 7]
```diff
@@ -1,31 +1,30 @@
...
+        return 200
+      }
     }
-
-    request(hub, function (err, res) {
-      if (err || res.statusCode !== 200) {
-        setTimeout(hasStarted, retryInterval);
-        return;
+  };
+  (async () => {
```
Comment (mgrybyk) [inline lib/check-started.js:19]: Can yo define async function outside and then call it here?

File: test/default-downloads-test.js [hunk 8]
```diff
@@ -13,18 +13,17 @@ var opts = {
...
+    try {
+      const res = await got(url,{retry:0});
+      if (res.statusCode >= 400) {
```
Comment (mgrybyk) [inline test/default-downloads-test.js:19]: I guess this can never happen as far as got throws an error in this case

**Review by mgrybyk — CHANGES_REQUESTED** [review]

**Comment by mgrybyk** [comment]
I'll pick this up, thanks for help!

**Comment by thomasschuiki** [comment]
Hi,

sorry I couldn't get the comments resolved in time. Glad I could help anyway! :)

**Pull request closed without merging**